File: backend/app/schemas/discovery.py
from typing import List, Optional

from pydantic import BaseModel

from app.schemas.paper import PaperSummary


class DiscoveryRequest(BaseModel):
    seed_ids: List[str]
    strategy: str = "co_citation"  # "co_citation" | "bibliographic_coupling"
    max_results: int = 30
    citing_sample_size: int = 100


class DiscoveryResult(BaseModel):
    paper: PaperSummary
    score: float
    overlap_seeds: int
    reason: Optional[str] = None


class DiscoveryResponse(BaseModel):
    strategy: str
    seed_count: int
    results: List[DiscoveryResult]

File: frontend/src/api/discovery.ts
import api from './client';
import type { PaperSummary } from './search';

export interface DiscoveryResult {
  paper: PaperSummary;
  score: number;
  overlap_seeds: number;
  reason: string | null;
}

export interface DiscoveryResponse {
  strategy: string;
  seed_count: number;
  results: DiscoveryResult[];
}

export async function runMultiSeedDiscovery(
  seedIds: string[],
  strategy: string = 'co_citation',
  maxResults: number = 30,
): Promise<DiscoveryResponse> {
  const resp = await api.post<DiscoveryResponse>('/discovery/multi-seed', {
    seed_ids: seedIds,
    strategy,
    max_results: maxResults,
  });
  return resp.data;
}
